credentials/sts/sts_test.go

```diff
@@ -765,3 +770,175 @@ func (s) TestTokenInfoFromResponse(t *testing.T) {
+	if err != nil {
+		t.Fatalf("did not read tlsCA: %v", err)
+	}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(serverCA)
+
+	validServerTLSConfig := &tls.Config{
+		ServerName: "grpc.test.example.com",
+		RootCAs:    caCertPool,
+	}
+
+	validSTSClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				ServerName: "sts.test.example.com",
+				RootCAs:    caCertPool,
+			},
+		}}
+
+	invalidSTSClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				ServerName: "bar.example.com",
+				RootCAs:    caCertPool,
+			},
+		}}
+
+	defaultSTSClient := http.DefaultClient
+
+	tests := []struct {
+		name          string
+		grpcTLSConfig *tls.Config
+		stsClient     *http.Client
+		wantErr       bool
+	}{
+		{
+			name:          "Test STS Server with custom RootCAs",
+			grpcTLSConfig: validServerTLSConfig,
+			stsClient:     validSTSClient,
+			wantErr:       false,
+		},
+		{
+			name:          "Test STS Server with defaultClient",
+			grpcTLSConfig: validServerTLSConfig,
+			stsClient:     defaultSTSClient,
+			wantErr:       true,
+		},
+		{
+			name:          "Test STS Server with nil client",
+			grpcTLSConfig: validServerTLSConfig,
+			stsClient:     nil,
+			wantErr:       true,
+		},
+		{
+			name:          "Test STS Server with incorrect SNI",
+			grpcTLSConfig: validServerTLSConfig,
+			stsClient:     invalidSTSClient,
+			wantErr:       true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			stsCreds, err := NewCredentials(Options{
+				TokenExchangeServiceURI: fmt.Sprintf("%s/token", stsServer.URL),
+				Resource:                exampleResource,
+				Audience:                exampleAudience,
+				Scope:                   testScope,
+				SubjectTokenPath:        testdata.Path("x509/server_ca_cert.pem"), // just read any file data and pretend its the token
+				SubjectTokenType:        subjectTokenType,
+				RequestedTokenType:      requestedTokenType,
+				HTTPClient:              test.stsClient,
+			})
+			if err != nil {
+				t.Fatalf("error creating STS Credentials: (%+v)", err)
+			}
+
+			creds := credentials.NewTLS(test.grpcTLSConfig)
+			conn, err := grpc.Dial(grpcListener.Addr().String(), grpc.WithTransportCredentials(creds), grpc.WithPerRPCCredentials(stsCreds))
+			if err != nil {
+				t.Fatalf("unexpected error on Dial (%+v) for test %s", err, test.name)
+			}
+			defer conn.Close()
+
+			c := testgrpc.NewTestServiceClient(conn)
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+			rpcResp, err := c.UnaryCall(ctx, &testpb.SimpleRequest{
+				FillUsername: true,
+			})
+			if (err != nil) != test.wantErr {
+				t.Fatalf("UnaryCall() returned error: %v, wantErr: %v", err, test.wantErr)
+			}
+			// got an error and expected one
+			if err != nil && test.wantErr {
+				return
+			}
+			t.Logf("SimpleResponse Username %s", rpcResp.Username)
```

Comment:
Are these still required?

Reply:
no, not needed; i just looking for full end-to end response.  i'll remove it